etl/dw/f_geracao_distribuida.py

```
@@ -0,0 +1,223 @@
+            schema=self.schema,
+            if_exists='append',
+        )
+
+    def truncate(self):
+        self._conn_output.execute(f'--sql TRUNCATE TABLE {self.schema}.{self.table_name} ;'[5:-1])
+
+    def run(self):
+        self.before_run()
+        self.extract()
+        self.truncate()
+        for self.df_extract in self.itercursor():
```

Ruff (0.12.2)

Loop control variable `self` overrides iterable it iterates

(B020)